blog: coder.ddev.com (#569)
ddev/ddev.com: src/content/blog/coder-ddev-com-announcement.md

:::warning[Experimental Service]
This is an experimental service with no guarantees of data retention, uptime, or long-term availability. The future of its maintenance and sustainability is uncertain. Do not store irreplaceable work here without pushing it to Git. Treat it as a convenience, not a platform to depend on.
:::

Want a quick overview? Watch the 6-minute intro video starting from the very beginning:

<div class="video-container">
<iframe width="560" height="315" src="https://www.youtube.com/embed/GGzh3yJo30E?si=q1BbZ4ZDBkosNrNr" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
</div>

## Table of Contents

## How It Works

coder.ddev.com runs on [Coder](https://coder.com), an open-source platform for remote development environments. Each workspace is an isolated container (using the [Sysbox](https://github.com/nestybox/sysbox) runtime for secure Docker-in-Docker) with DDEV, Docker, and VS Code pre-installed.

> Your files persist on a remote volume across workspace restarts. When you delete a workspace, the data is gone — so push your work to Git before deleting. (But until you delete the workspace, or it's garbage-collected, your work persists. We don't have an implementation for garbage collection yet...)

Reviewer: ```suggestion
Your files persist on a remote volume across workspace restarts. When you delete a workspace, the data is gone — so push your work to Git before deleting. But until you delete the workspace, or it's garbage-collected (not yet implemented), your work persists..
```

The source code for the templates and Docker image is at [github.com/ddev/coder-ddev](https://github.com/ddev/coder-ddev). Other projects can use this and deploy their own fully-DDEV-capable Coder instances.

## Getting Started

### 1. Log In with GitHub

> Go to [coder.ddev.com](https://coder.ddev.com) and click **Login with GitHub**. No separate account needed. `coder.ddev.com` gets only your email address.

Reviewer 2: ```suggestion
Go to [coder.ddev.com](https://coder.ddev.com) and click **Login with GitHub**. No separate account needed. `coder.ddev.com` is only allowed to read out your email address.
```
i wouldnt necessary use the term "get" that feels like way more intentional that they want to get hold of your email address and now they are allowed to do shady things. that kind of words when used always make me cautious.

<div class="video-container">
<iframe width="560" height="315" src="https://www.youtube.com/embed/RxMpI4WgX-A?si=RTfezKbVQ61Erw_G" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
</div>

### freeform

The freeform template adds Traefik routing integration so your DDEV project and services like Mailpit get stable subdomain URLs (no port numbers). After creating a workspace, run `ddev coder-setup` once in your project directory, then `ddev start`. Routing updates automatically on every start.

## The Drupal Issue Picker

One of the most useful features for Drupal contributors is the **Drupal Issue Picker** at [start.coder.ddev.com/drupal-issue](https://start.coder.ddev.com/drupal-issue).

![Drupal Core Issue Picker — enter an issue URL or number to launch a pre-configured workspace](/img/blog/2026/03/drupal-issue-picker.png)

Paste any drupal.org issue URL (for example, `https://www.drupal.org/project/drupal/issues/3568144`) and the picker launches a drupal-core workspace with:

> - The correct Drupal version (10.x, 11.x, or main) detected from the issue

Author: You can open anything you want, you can check out any issue fork you want, it's just ddev.

- The issue fork branch already checked out
- Composer dependencies resolved

This replaces the workflow that DrupalPod (Gitpod-based) provided for contribution days. You can hand someone an issue URL, they paste it into the picker, and within 30 seconds they have a working environment with the issue branch ready.

Demonstrating this from start to finish in about 6 minutes:

<div class="video-container">
<iframe width="560" height="315" src="https://www.youtube.com/embed/2BCOl86SFvw?si=TmUqzEUrjHZMqlA8" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>
</div>

## Development Tools

### VS Code for Web

> VS Code for Web runs in the browser and supports most extensions. You can install extensions from the marketplace, configure settings, and use the integrated terminal — all without installing anything locally.

Reviewer: ```suggestion
"VS Code Web" runs in the browser and supports most extensions. You can install extensions from the marketplace, configure settings, and use the integrated terminal — all without installing anything locally.
```

### Desktop VS Code

For a native IDE experience, install the [Coder CLI](https://github.com/coder/coder/releases) and run:

```bash
coder login https://coder.ddev.com
coder config-ssh
```

> Then use VS Code's **Remote-SSH** extension to connect to `coder.<workspace-name>`. All VS Code features work, including extensions that require a desktop environment.

Reviewer: ```suggestion
Coder uses the [**Remote-SSH** extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-ssh) to connect to `coder.<workspace-name>`. All VS Code features work, including extensions that require a desktop environment.
```

### Xdebug

Xdebug works in Coder workspaces the same way as local DDEV:

```bash
ddev xdebug on
```

> With desktop VS Code connected via Remote-SSH, configure your `launch.json` for remote debugging as you would with any remote PHP interpreter. With VS Code for Web, use the browser debugger integration.

Reviewer: ```suggestion
With "VS Code desktop" connected via Remote-SSH, configure your `launch.json` for remote debugging as you would with any remote PHP interpreter. With "VS Code Web", use the browser debugger integration.
```
